opm/input/eclipse/Schedule/UDQ/UDQAssign.hpp

```diff
@@ -17,89 +17,127 @@
         {
-            serializer(input_selector);
-            serializer(rst_selector);
-            serializer(value);
-            serializer(report_step);
+            serializer(this->input_selector);
+            serializer(this->rst_selector);
+            serializer(this->numbered_selector);
+            serializer(this->value);
+            serializer(this->report_step);
         }
     };
 
-    UDQAssign();
-    UDQAssign(const std::string& keyword, const std::vector<std::string>& selector, double value, std::size_t report_step);
-    UDQAssign(const std::string& keyword, const std::unordered_set<std::string>& selector, double value, std::size_t report_step);
+    UDQAssign() = default;
+    UDQAssign(const std::string&              keyword,
+              const std::vector<std::string>& selector,
+              double                          value,
+              std::size_t                     report_step);
+
+    UDQAssign(const std::string&                     keyword,
+              const std::unordered_set<std::string>& selector,
+              double                                 value,
+              std::size_t                            report_step);
+
+    UDQAssign(const std::string&                              keyword,
+              const std::vector<UDQSet::EnumeratedWellItems>& selector,
+              double                                          value,
+              std::size_t                                     report_step);
+
+    UDQAssign(const std::string&                         keyword,
+              std::vector<UDQSet::EnumeratedWellItems>&& selector,
+              double                                     value,
+              std::size_t                                report_step);
 
     static UDQAssign serializationTestObject();
 
     const std::string& keyword() const;
     UDQVarType var_type() const;
-    void add_record(const std::vector<std::string>& selector, double value, std::size_t report_step);
-    void add_record(const std::unordered_set<std::string>& rst_selector, double value, std::size_t report_step);
+
+    void add_record(const std::vector<std::string>& selector,
+                    double                          value,
+                    std::size_t                     report_step);
+
+    void add_record(const std::unordered_set<std::string>& rst_selector,
+                    double                                 value,
+                    std::size_t                            report_step);
+
+    void add_record(const std::vector<UDQSet::EnumeratedWellItems>& selector,
+                    double                                          value,
+                    std::size_t                                     report_step);
+
+    void add_record(std::vector<UDQSet::EnumeratedWellItems>&& selector,
+                    double                                     value,
+                    std::size_t                                report_step);
+
```

Comment:
Any reason why this one special? The other options don't support moving.

Reply:
> Any reason why this one special?

Mostly because if we actually hit this constructor, then the `selector` is **typically** going to be rather large.  I wanted to optimise the copy operation in that case.